services/repository/files/upload.go

```diff
@@ -107,10 +107,12 @@ func UploadRepoFiles(ctx context.Context, repo *repo_model.Repository, doer *use
 	}
 
 	var attributesMap map[string]*attribute.Attributes
-	if setting.LFS.StartServer {
+	if setting.LFS.StartServer && hasOldBranch {
```

Comment:
Is the `hasOldBranch` still necessary?

Reply:
OK, I am sure that we should not check `hasOldBranch`, because the "upload" should work on an empty repo, which doesn't have "old branch"

Reply:
I think it doesn't matter atm, an empty repo doesn't have a .gitattributes file that could specify if files should be stored in LFS, so the `attributesMap` won't contain relevant information.
All uploaded files are added to git after the attribute check is done, maybe it's an idea (for another PR) to add a potentially uploaded .gitattributes in the temp repo before checking the attributes